keras_cv/models/generative/stable_diffusion/stable_diffusion.py

```diff
@@ -111,6 +112,35 @@ def __init__(self, img_height=512, img_width=512, jit_compile=False):
         self.diffusion_model.load_weights(diffusion_model_weights_fpath)
         self.decoder.load_weights(decoder_weights_fpath)
 
+    def add_tokens(self, tokens):
+        added_tokens = self.tokenizer.add_tokens(tokens)
+        new_vocab_size = self.text_encoder.vocab_size + added_tokens
+
+        old_token_weights = self.text_encoder.embedding.token_embedding.get_weights()
+        old_position_weights = (
+            self.text_encoder.embedding.position_embedding.get_weights()
+        )
+        if len(old_token_weights) > 1:
+            # this should never happen, but we should warn users if something unexpected
+            # changes upstream.
+            raise ValueError("Invalid old_token_weights or old_position_weights")
+
+        old_token_weights = old_token_weights[0]
+        new_weights = np.mean(old_token_weights, axis=0)
+        new_weights = np.expand_dims(new_weights, axis=0)
+        new_weights = np.repeat(new_weights, added_tokens, axis=0)
```

Comment:
You should add noise to avoid initializing multiple embeddings to the same value